### DISCUSSION TITLE
vercel dev / python runtime / windows 10

### DISCUSSION THREAD

**Discussion by LeandroAlberti** [vercel/vercel#7519] [Help] [answered]
I'm trying to start a dev environment with python runtime.

I have this file only:

api/date.py:
```py
from http.server import BaseHTTPRequestHandler
from datetime import datetime

class handler(BaseHTTPRequestHandler):

  def do_GET(self):
    self.send_response(200)
    self.send_header('Content-type', 'text/plain')
    self.end_headers()
    self.wfile.write(str(datetime.now().strftime('%Y-%m-%d %H:%M:%S')).encode())
    return
```

Then used "vercel dev" command.

When i tried in my browser:
    localhost:3000/api/date.py

received this message on browser:
    Command failed: pip3 install --disable-pip-version-check --target . werkzeug==1.0.1

and in shell:

> Vercel CLI 24.0.0
> Ready! Available at http://localhost:3000
> Building @vercel/python@latest:api/date.py
Installing required dependencies...
Failed to run "pip3 install --disable-pip-version-check --target . werkzeug==1.0.1"
Error: Command failed: pip3 install --disable-pip-version-check --target . werkzeug==1.0.1
    at makeError (C:\Users\_\AppData\Roaming\npm\node_modules\vercel\node_modules\@vercel\python\dist\index.js:283:9)
    at C:\Users\_\AppData\Roaming\npm\node_modules\vercel\node_modules\@vercel\python\dist\index.js:387:16  
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async pipInstall (C:\Users\_\AppData\Roaming\npm\node_modules\vercel\node_modules\@vercel\python\dist\index.js:3777:9)
    at async Object.installRequirement (C:\Users\_\AppData\Roaming\npm\node_modules\vercel\node_modules\@vercel\python\dist\index.js:3796:5)
    at async Object.build (C:\Users\_\AppData\Roaming\npm\node_modules\vercel\node_modules\@vercel\python\dist\index.js:3584:5)
    at async processMessage (C:\Users\\_\AppData\Roaming\npm\node_modules\vercel\dist\builder-worker.js:39:18)
{
  code: 3221225477,
  stdout: '',
  stderr: '',
  failed: true,
  signal: null,
  cmd: 'pip3 install --disable-pip-version-check --target . werkzeug==1.0.1',
  timedOut: false,
  killed: false
}

I have Python 3.10.2 and pip 22.0.3 installed

How can i proceed?

**Comment by ari-motors-team** [comment, marked as answer]
At the moment only python 3.9 is supported. However, I am still struggling to to the python 3.9 working with `vercel dev`

**Reply by LeandroAlberti** [reply]
Got it, I think this should receive attention to keep it up to date

**Reply by dmarcucci** [reply]
For some reason, I'm able to use a later version (3.10.2) no problem when I run it from a standard command prompt. But when trying from a terminal in my IDE scoped to my virtual environment with the same version, I am also having this issue. 

Using 3.9 in a virtual environment, however, also works for me.

**Reply by kuldeepvarma7413** [reply]
Command failed: pip3.9 install --disable-pip-version-check --target . --upgrade -r /vercel/path0/requirements.txt error: subprocess-exited-with-error

i am getting this while deploying django project on vercel. help me...
